src/features/dashboard/components/ApiTokenTable/__tests__/index.test.tsx

```diff
@@ -72,76 +68,31 @@ describe('DeleteTokenDialog', () => {
     expect(token_row).toBeInTheDocument();
   });
 
-  it('Shows the dialog when pressing the delete button', async () => {
-    const delete_button = await screen.findByTestId('dt_delete_button_0');
-    expect(delete_button).toBeInTheDocument();
+  it('Should have the table headers and the header description', async () => {
+    const header_text = await screen.findByText(/API token manager/i);
+    expect(header_text).toBeInTheDocument();
 
-    await act(async () => {
-      await userEvent.click(delete_button);
-    });
-
-    const delete_modal = await screen.findByText(/Are you sure you want to delete this token?/i);
-    expect(delete_modal).toBeInTheDocument();
+    const header_descr = await screen.findByText(/Access all your API token details here/i);
+    expect(header_descr).toBeInTheDocument();
   });
 
-  it('Should close the dialog when pressing the close button', async () => {
-    const delete_button = await screen.findByTestId('dt_delete_button_0');
+  it('Shows the dialog when pressing the delete button', async () => {
+    const delete_button = await screen.findByTestId('delete-token-button');
+    await userEvent.click(delete_button);
     expect(delete_button).toBeInTheDocument();
-
-    await act(async () => {
-      await userEvent.click(delete_button);
-    });
-
-    const delete_modal = await screen.findByText(/Are you sure you want to delete this token?/i);
-    expect(delete_modal).toBeInTheDocument();
-
-    // Using test id provided from UI library
-    const close_button = await screen.findByTestId('close-button');
-    await act(async () => {
-      await userEvent.click(close_button);
-    });
-
-    expect(delete_modal).not.toBeInTheDocument();
   });
 
-  it("Should close the dialog when pressing the 'No, keep it' button", async () => {
-    const delete_button = await screen.findByTestId('dt_delete_button_0');
-    expect(delete_button).toBeInTheDocument();
-
-    await act(async () => {
-      await userEvent.click(delete_button);
-    });
-
-    const delete_modal = await screen.findByText(/Are you sure you want to delete this token?/i);
-    expect(delete_modal).toBeInTheDocument();
-
-    const cancel_button = await screen.findByRole('button', { name: 'Cancel' });
-    await act(async () => {
-      await userEvent.click(cancel_button);
-    });
-
-    expect(delete_modal).not.toBeInTheDocument();
+  it('Should delete the token or cancel and close the modal', async () => {
+    const actions_token = await screen.findByTestId('token-action-cell');
+    await userEvent.click(actions_token);
+    expect(actions_token).toBeInTheDocument();
```

Comment:
We are performing an action here so we should expect something that should occur after performing the action.

Reply:
Combined this one and the above one just like how it was done in the apps table test case.